### DISCUSSION TITLE
How to clean the cache?

### DISCUSSION THREAD

**Discussion by nrbnlulu** [astral-sh/rye#990] [Q&A]
I figure that rye uses some global package cache. If I accidentally modify some package sources I would like to distinct the cache is it possible?

**Comment by failable** [comment]
Same question

**Comment by nrbnlulu** [comment]
Seems like it is possible in `uv`. `rye` don't expose \ delegate the needed command to uv ATM see https://github.com/nrbnlulu/rye/blob/43ac3a3ebfe2fd647e7b9a9fca723eeb37194b7d/rye/src/uv.rs#L542

and 
https://github.com/astral-sh/uv/blob/42c3bfa351cc2cc3d0325da20f6f5d97f2fcf7be/crates/uv/src/cli.rs#L663

I am willing to work on this if some guidance would be given.

**Comment by bluss** [comment]
Just to give more information, not a solution, UV listens to a lot of environment variables which you can use even with rye+uv, for example UV_NO_CACHE=true rye sync -f   would rebuild the environment without any cache. UV has multiple caches (downloads, git, wheels, etc..)

The environment variables are documented here for the time being https://github.com/astral-sh/uv?tab=readme-ov-file#environment-variables

Keeping in mind like you said that Rye's bundled uv trails the uv project by some releases, usually.
